src/core/config/option.h

```diff
@@ -90,6 +90,11 @@ class Option : public IOption {
         return *this;
     }
 
+    std::unique_ptr<OptValue> GetOptValue() override {
+        return std::make_unique<OptValue>(
+                OptValue{std::type_index(typeid(T)), boost::any(*value_ptr_)});
```

Comment:
it is needed to keep std::type_index in map, because if you try to add new OptValue object in map without pointer it throws an error (there is no constructor for std::type_index)

https://stackoverflow.com/questions/46197573/use-stdtype-index-as-value-in-a-map